speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/FileDataStorage.h

```diff
@@ -46,7 +46,17 @@ class FileDataStorage
     int Partition(float* p_values, int p_start, int p_end);
 
 public:
-    std::experimental::filesystem::path Initialize(
+    /// @brief Add the new value to the array in the correct compression step
+    /// @param p_values Array with values from the current compression step
+    /// @param p_value The new value of this time step for the variable
+    /// @param p_compressionStep The current compression step
+    template <typename TNumber>
+    void AddToArray(TNumber* p_values, TNumber p_value, int p_compressionStep) const
+    {
+        p_values[p_compressionStep] = p_value;
+    }
```

Comment:
Why was this definition moved from the cpp to the header file?

Reply:
It did not want to build on Linux as the definition of AddToArray is not available for tests, I was not able to use an explicit definition in the GCC compiler so I converted it to an implicit definition